api4/config_test.go

```diff
@@ -471,6 +471,8 @@ func TestUpdateConfigDiffInAuditRecord(t *testing.T) {
 	})
 	require.Equal(t, timeoutVal+1, *cfg.ServiceSettings.ReadTimeout)
 
+	require.NoError(t, logFile.Sync())
+
 	// Forcing a flush before attempting to read log's content.
 	err = th.Server.Log.Flush(context.Background())
```

Comment:
I think the issue here is that flush doesn't close the file like shutdown does.  There is a bug in the version of Logr used here that doesn't do an explicit flush during shutdown, meaning you should flush then shutdown.  This is fixed in Focalboard and I'll be bringing that to server next sprint.

Reply:
Hmm .. so do you want me to hold off from making this change?

Reply:
No, the change is good.  I just don't know what file.Sync is going to do on a open file.  I'm suggesting you call shutdown after flush, then sync.  That should cover all bases and will continue to work in the future.

Reply:
But shutdown closes the file descriptor. So the kernel can reclaim it for other uses.

[Sync](https://pkg.go.dev/os#File.Sync) will just flush the contents to disk. Internally it just calls `fsync`  on the `fd`. So calling sync on an open file should be fine I believe.

And since `th.Teardown` already calls Shutdown, I think that should be fine. As long as the contents are synced to disk, the test should pass.

Reply:
Ok, sounds good.